python/pyiceberg/typedef.py

```diff
@@ -72,34 +71,9 @@ class StructProtocol(Protocol):  # pragma: no cover
     """A generic protocol used by accessors to get and set at positions of an object"""
 
     @abstractmethod
-    def get(self, pos: int) -> Any:
+    def __getitem__(self, pos: int) -> Any:
         ...
 
     @abstractmethod
-    def set(self, pos: int, value: Any) -> None:
+    def __setitem__(self, pos: int, value: Any) -> None:
         ...
-
-
-class Record(StructProtocol):
```

Comment:
I moved these to `iceberg_base_model.py` because of circular references. I'm also completely fine with merging `iceberg_base_model.py` into `typedef.py`. Also removes one file.

Reply:
I think it makes sense to put everything into `typedef`.